### DISCUSSION TITLE
Which file types have Nerd Font icons?

### DISCUSSION THREAD

**Discussion by bluelephant825** [pls-rs/pls#93] [Help Q&A] [answered]
Hi Dhruv, I noticed that certain file types, e.g. html, css or js, don't get an icon when I perform a pls in the terminal. Therefore, I wanted to ask you if there's a list somewhere of the file types you have taken care of, i.e. for which a Nerd Font icon has been associated.

**Comment by dhruvkb** [comment, marked as answer]
This is the list of icons names built into `pls`, each with its associated Nerd Fonts icon. (If you see just squares below, you will need to adjust your browser's monospace font.)

https://github.com/pls-rs/pls/blob/5fc0373e1cfd2f7d3981c9c9f25e93812cbc3f5e/src/config/conf.rs#L41-L75

This is the list of specs built into `pls`, each with its associated icon name.

https://github.com/pls-rs/pls/blob/5fc0373e1cfd2f7d3981c9c9f25e93812cbc3f5e/src/config/conf.rs#L78-L106

I can understand how this can have been communicated better in the docs.

| Pattern | Icon |
|-|-|
| `\.sh$` | shell |
| `\.rs$` | rust |
| `\.(txt\|rtf)$` | text |
| `\.mdx?$` | markdown |
| `\.ini$` | config |
| `\.(json\|toml\|yml\|yaml)$` | json |
| `\.(jpg\|jpeg\|png\|svg\|webp\|gif\|ico)$` | image |
| `\.(mov\|mp4\|mkv\|webm\|avi\|flv)$` | video |
| `\.(mp3\|flac\|ogg\|wav)$` | audio |
| `^\.env\b` | env |
| `^README\b` | book |
| `^LICENSE\b` | law |
| `docker-compose.*\.yml$` | container |
| `Dockerfile` | container |
| `^\.DS_Store$` | apple |
| `^\.pls\.yml$` | pls |
| `^\.git$` | git |
| `^\.gitignore$` | git |
| `^\.github$` | github |
| `^src$` | source |
| `^(justfile\|Makefile)$` | runner |
| `^Cargo\.toml$` | package |
| `^Cargo\.lock$` | loc |
| `^rustfmt.toml$` | broom |

The icons for web development are not built in and will need to be added by the user. You can refer to the [starter configs in the `pls` documentation](https://pls.cli.rs/cookbooks/starters/) for that.

**Reply by dhruvkb** [reply]
I also just added a starter config for web dev projects involving HTML, CSS and some CSS pre-processors.

**Comment by bluelephant825** [comment]
Beautiful, Dhruv! Thank you so much. I was actually having a look at the Material Design Files Icons and thought that [this file](https://github.com/PKief/vscode-material-icon-theme/blob/main/src/icons/fileIcons.ts) could be used to generate the Specs. The difficulty I'm having is to automatically associate an Nerd Font Icon Name to an icon name. I don't think there's an API that I could use.

Btw, can the icon always be left blank as shown below, as an example (in other words, does it suffice to have the name nf-seti-json specified):

`"json"         => " ", // nf-seti-json `

![File Icons](https://github.com/pls-rs/pls/assets/4818315/11e43c0a-349e-4ea2-bff1-22126feb7d45)

**Reply by dhruvkb** [reply]
`pls` ships with a limited set of specs so that the app can be performant. Matching specs to node names is a time-consuming expensive operation so the fewer the specs the faster `pls` runs. That's why `pls` recommends making a `.pls.yml` file inside your project repo so that the specs are localised to that project.

You can refer to the [Nerd Fonts cheat sheet](https://www.nerdfonts.com/cheat-sheet) for the list of available icons. It's much more limited compared to the icons in an icon pack because
- they're packaged in a font
- they're not changed and updated as often
- they need to be monochromatic

The VS Code icon themes will definitely be much more comprehensive and suitable for UI-based applications.

As for the `.pls.yml` file, the icon glyph is needed. The text after the `#` is just a comment, that can be skipped. So to your question, no, the icon cannot be skipped but the name can.

```yaml
icons:
  json: ""
```
